web/src/timestamp_width_handler.js

```diff
@@ -0,0 +1,231 @@
+        maxtimestamp12_hr: "ප.ව. 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "Sinhala",
+    },
+    fi: {
+        maxtimestamp12_hr: "11:59 ip.",
+        maxtimestamp24_hr: "23:59",
+        name: "Suomi",
+    },
+    sv: {
+        maxtimestamp12_hr: "11:59 em",
+        maxtimestamp24_hr: "23:59",
+        name: "Svenska",
+    },
+    tl: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Tagalog",
+    },
+    vi: {
+        maxtimestamp12_hr: "11:59 CH",
+        maxtimestamp24_hr: "23:59",
+        name: "Tiếng Việt",
+    },
+    tr: {
+        maxtimestamp12_hr: "ÖS 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "Türkçe",
+    },
+    be: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Беларуская",
+    },
+    bg: {
+        maxtimestamp12_hr: "11:59 ч. сл.об.",
+        maxtimestamp24_hr: "23:59 ч.",
+        name: "Български",
+    },
+    ru: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Русский",
+    },
+    sr: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "Српски",
+    },
+    uk: {
+        maxtimestamp12_hr: "11:59 пп",
+        maxtimestamp24_hr: "23:59",
+        name: "Українська",
+    },
+    ar: {
+        maxtimestamp12_hr: "١١:٥٩ م",
+        maxtimestamp24_hr: "٢٣:٥٩",
+        name: "العربيّة",
+    },
+    fa: {
+        maxtimestamp12_hr: "۱۱:۵۹ بعدازظهر",
+        maxtimestamp24_hr: "۲۳:۵۹",
+        name: "فارسی",
+    },
+    hi: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "हिंदी",
+    },
+    ta: {
+        maxtimestamp12_hr: "பிற்பகல் 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "தமிழ்",
+    },
+    ml: {
+        maxtimestamp12_hr: "11:59 PM",
+        maxtimestamp24_hr: "23:59",
+        name: "മലയാളം",
+    },
+    ko: {
+        maxtimestamp12_hr: "오후 11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "한국어",
+    },
+    ja: {
+        maxtimestamp12_hr: "午後11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "日本語",
+    },
+    "zh-CN": {
+        maxtimestamp12_hr: "下午11:59",
+        maxtimestamp24_hr: "23:59",
+        name: "简体中文",
+    },
+};
+
+export function calculateFormattedTimeWidth() {
+    const locale = timerender.get_user_locale();
+    const localeData = candidate_timestamps[locale];
+    let formattedTimeWidth = 45; // Default width if locale is not found
+
+    if (localeData) {
+        // timestamp contains the string according to timeformat.
+        const timestamp = user_settings.twenty_four_hour_time
+            ? localeData.maxtimestamp24_hr
+            : localeData.maxtimestamp12_hr;
+
+        // logic for rendering timestamp in hidden div and calculating it's width.
+        const hiddenDiv = document.createElement("div");
+
+        hiddenDiv.style.visibility = "hidden";
```

Comment:
Your `<div>` shouldn't be a set of approximate styles; instead, it should take the same classes and structures necessary for the rendering of actual timestamps in the message area. That's a tricky proposition because of how our selectors are currently written, so you might be generating a more complicated piece of markup here (or putting up a prep commit/PR for simplifying the selectors around `.message_time`)

Reply:
One thing to keep in mind is we want this approach to work regardless of changes to the design; the code here should always reflect what's actually being rendered on screen.